contracts/FragToken.sol

```diff
@@ -1,13 +1,13 @@
 /* SPDX-License-Identifier: BUSL-1.1 */
 /* Copyright © 2021 Fragcolor Pte. Ltd. */
 
-pragma solidity ^0.8.7;
+pragma solidity >=0.8.0 <0.9.0;
 
 import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
-import "openzeppelin-solidity/contracts/token/ERC20/extensions/draft-ERC20Permit.sol";
 import "openzeppelin-solidity/contracts/access/Ownable.sol";
+import "openzeppelin-solidity/contracts/utils/cryptography/draft-EIP712.sol";
 
-contract FRAGToken is ERC20, ERC20Permit, Ownable {
+contract FRAGToken is ERC20, EIP712, Ownable{
```

Comment:
we we need to derive EIP712? Can't we use it as a library? Any security implications from inheriting should be double checked too.

Reply:
We were already indirectly deriving from it, since EIP712 is derived by ERC20Permit. I just did it directly since we do not need ERC20Permit (as of now, at least).
EIP712 is an official Openzeppelin contract, as ERC20. I assume security has already been assessed on their side. 
I can change to use it as library, if you prefer.

Reply:
My point is that EIP712 has **Nothing** to do with a ERC20... We don't really know what we are pulling. Also you removed `ERC20Permit` which I wanted.

Reply:
If we inherit for a single static/pure function it's super weird

Reply:
Also is it gas efficient? maybe you can find a better implementation?